internal/cli/cli.go

```diff
@@ -89,11 +93,28 @@ func NewCommandFactory() CommandFactory {
 	}
 }
 
+func (factory *commandFactory) configureTelemetry() error {
+	telemetryMode := factory.config.TelemetryMode
+	if telemetryMode == telemetry.ModeNil {
+		telemetryMode = factory.profile.GetTelemetryMode()
+	}
+	factory.tracker = telemetry.NewTracker(
+		telemetryMode,
+		factory.profile.GetUser().PublicAPIKey,
+		primitive.NewObjectID().Hex())
+	factory.profile.SetTelemetryMode(telemetryMode)
```

Comment:
i'd like to only call set/save here on the profile if it differs from the flagged value.  can you check/do that before we create the `factory.tracker`?